src/Model/TransportModel/tsp1apt1.f90

```diff
@@ -76,7 +77,11 @@ module GwtAptModule
     integer(I4B), pointer :: ibudcsv => null() !< unit number for csv budget output file
     integer(I4B), pointer :: ncv => null() !< number of control volumes
     integer(I4B), pointer :: igwfaptpak => null() !< package number of corresponding this package
-    real(DP), dimension(:), pointer, contiguous :: strt => null() !< starting feature concentration
+    integer(I4B), pointer :: idxprepak => null() !< budget-object index that precedes package-specific budget objects
+    integer(I4B), pointer :: idxlastpak => null() !< budget-object index of last package-specific budget object
```

Comment:
Unclear why we need these.

Reply:
These are part of a modification by @aprovost-usgs for the benefit of UZE. In formulating the matrix problem, the UZE balance equation is folded into the corresponding cell balance equation. In the subsequent budget calculations, the heat exchanged between the UZE feature and the cell (by virtue of their coming to thermal equilibrium) is "backed out" as the residual of the UZE budget terms. Therefore, the UZE-package-specific budget calculations, which include this residual calculation as the final step, need to come after the generic, APT-based budget calculations. Although (currently) only UZE should require this reordering, it is simplest to do it that way for all advanced packages and should pose no problem.  `idxprepak` and `idxlastpak` are indices that are maintained so that the package specific calculations can be initially skipped and then done at the end.

Reply:
Alternatively, UZE could have overridden parent routines and stored these indices on its own.  I'm fine with whatever way you want to go, especially if you think this is general and possibly useful beyond UZE in the future.

Reply:
@langevin-usgs Agree there are several ways this could be done. The bottom-line requirement is that all budget terms be in place before UZE calculates its residual.

-- The way it's done now, duplication of code is minimized, but it's admittedly a bit klugey to skip the package-specific calls and then comeback to them at the end.

-- Riffing off of the way it's done now, the kluginess could be eliminated by maybe reordering the indexing of the budget terms in APT so the package-specific terms come last. Haven't thought through the possible side effects.

-- As you suggested, UZE could override and handle all the budget calculations, doing them in whatever order it needs, at the expense of some code duplication.

-- Could break out the residual calculation as a separate step at the end. This is actually a more generally applicable approach, since the alternatives above will work only as long as UZE is the only APT package that needs to calculate its residual. So the idea would be to complete (in any order) all the generic and package-specific budget calculations other than the residuals, then at the end call package-specific residual-calculating routines (which would do nothing for most packages). Not sure if we'll ever have another residual-dependent package like UZE, but maybe it would be best to go ahead set it up that way anyway? The cost would be an additional subroutine baked into the APT infrastructure. (It's been a while since I looked deeply at this part of the code, but I believe it would be ok to have multiple residual-dependent packages, since the residual represents an "internal" transfer within each package and shouldn't affect other packages.)

Happy to do it whatever way is best for MF6 in the long run.